File: backend/src/YouG.Domain/Entities/Group.cs
using YouG.Domain.Common;

namespace YouG.Domain.Entities;

public class Group : Entity
{
    public required string Name { get; set; }
    public string? Description { get; set; }
    public string? PictureUrl { get; set; }
    public required Guid CreatedByUserId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}

File: mobile/lib/features/groups/domain/entities/group.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'group.freezed.dart';

@freezed
abstract class Group with _$Group {
  const factory Group({
    required String id,
    required String name,
    String? description,
    String? pictureUrl,
    required String createdByUserId,
    required int memberCount,
    required DateTime createdAt,
  }) = _Group;
}
